pkg/api/kptfile/v1alpha2/types.go

```diff
@@ -0,0 +1,201 @@
+	Email string `yaml:"email,omitempty"`
+
+	// License is the package license
+	License string `yaml:"license,omitempty"`
+
+	// Version is a logical package version (ignored by kpt)
+	Version string `yaml:"version,omitempty"`
+
+	// Tags enables humans and tools to attach arbitrary package metadata.
+	Tags []string `yaml:"tags,omitempty"`
+
+	// Man is the path to documentation about the package
+	Man string `yaml:"man,omitempty"`
+
+	// ShortDescription contains a short description of the package.
+	ShortDescription string `yaml:"shortDescription,omitempty"`
+}
+
+// Subpackages declares a local or remote subpackage.
+type Subpackage struct {
+	// Name of the immediate subdirectory relative to this Kptfile where the suppackage
+	// either exists (local subpackages) or will be fetched to (remote subpckages).
+	// This must be unique across all subpckages of a package.
+	LocalDir string `yaml:"localDir,omitempty"`
+	// Whether a subpackage is local or remote is determined by whether Upstream is specified.
+	// Upstream is a reference to where the package is fetched from.
+	Upstream Upstream `yaml:"upstream,omitempty"`
+}
+
+// Git contains information on the origin of packages fetched from a git repository.
+type Git struct {
+	// Commit is the git commit that the package was fetched at
+	Commit string `yaml:"commit,omitempty"`
+
+	// Repo is the git repository the package was cloned from.  e.g. https://
+	Repo string `yaml:"repo,omitempty"`
+
+	// Directory is the sub directory of the git repository that the package was cloned from
+	Directory string `yaml:"directory,omitempty"`
+
+	// Ref is the git ref the package was cloned from
+	Ref string `yaml:"ref,omitempty"`
+}
+
+// Pipeline declares a pipeline of functions used to mutate or validate resources.
+type Pipeline struct {
+	//  Sources defines the source packages to resolve as input to the pipeline. Possible values:
+	//  a) A slash-separated, OS-agnostic relative package path which may include '.' and '..' e.g. './base', '../foo'
+	//     The source package is resolved recursively.
+	//  b) Resources in this package using '.'. Meta resources such as the Kptfile, Pipeline, and function configs
+	//     are excluded.
+	//  c) Resources in this package AND all resolved subpackages using './*'
+	//
+	// Resultant list of resources are ordered:
+	// - According to the order of sources specified in this array.
+	// - When using './*': Subpackages are resolved in alphanumerical order before package resources.
+	//
+	// When omitted, defaults to './*'.
+	// Sources []string `yaml:"sources,omitempty"`
+
+	// Following fields define the sequence of functions in the pipeline.
+	// Input of the first function is the resolved sources.
+	// Input of the second function is the output of the first function, and so on.
+	// Order of operation: mutators, validators
+
+	// Mutators defines a list of of KRM functions that mutate resources.
+	Mutators []Function `yaml:"mutators,omitempty"`
```

Comment:
Shouldn't this be []Mutator which in turn contains Function definition? Where can we specify filters here ? And additional metadata which is specific to scoping?

Reply:
Bunch of different questions here:

"Shouldn't this be []Mutator which in turn contains Function definition?"

Elaborate? Example Pipeline: go/kpt-pipeline-declaration

"Where can we specify filters here ?"

Per PR description, this still need to be finalized and added to the schema. But it will be an additive change.

"And additional metadata which is specific to scoping?"

Elaborate? "scoping" is generally an overloaded word.

Reply:
scoping: Where can a user specify filters to target only specific packages ? 

this still need to be finalized and added to the schema => got it.